server/jamovi/server/tests/test_dataset.py

```diff
@@ -0,0 +1,50 @@
+
+from pytest import fixture
+
+from tempfile import TemporaryDirectory
+from os import path
+
+from jamovi.core import DataSet
+from jamovi.core import MemoryMap
+from jamovi.core import ColumnType
+from jamovi.core import DataType
+from jamovi.core import MeasureType
+
+
+@fixture
+def temp_dir():
+    with TemporaryDirectory() as temp:
+        yield temp
+
+
+@fixture
+def memory_map(temp_dir: str):
+    temp_file = path.join(temp_dir, 'fred.mm')
+    mm = MemoryMap.create(temp_file)
+    yield mm
+    mm.close()
+
+
+@fixture
+def dataset(memory_map):
+    return DataSet.create(memory_map)
+
+
+def test_dataset(dataset):
+
```

Comment:
It's good practice to only test a single thing/behavior in a single test. What helps mee keep a test focussed is by adding bdd style comments in the form:
```suggestion
# GIVEN a dataset
# WHEN I add a number of columns to this dataset
# THEN the column count is set to the correct value
# AND the row count is set to the correct value
```
This also helps with reading the test at a later point (and maybe refactoring it if it breaks)